### PR TITLE
Migrate Helm Chart to OSDK v1.3.0

### PR DESCRIPTION
**Pull request by SylwiaBrant**
Migrated Helm Chart to Operator SDK v1.3.0.
Added e2e tests, also to GitHub Actions. 
Fixed Makefile goals.

### CODE REVIEW

File: chart/jenkins-operator/templates/leader_election_role_binding.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+kind: RoleBinding
+metadata:
+  name: leader-election-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: leader-election-role
+subjects:
+- kind: ServiceAccount
+  name: jenkins-operator
```
Comment (SylwiaBrant) [inline chart/jenkins-operator/templates/leader_election_role_binding.yaml:12]: Auto-generated from OSDK.

File: chart/jenkins-operator/templates/leader_election_role.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
```
Comment (SylwiaBrant) [inline chart/jenkins-operator/templates/leader_election_role.yaml:28]: Auto-generated from OSDK.

File: config/rbac/role.yaml [hunk 3]
```diff
@@ -1,119 +1,172 @@
...
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
+kind: Role
 metadata:
   creationTimestamp: null
-  name: manager-role
+  name: jenkins-operator
 rules:
-- apiGroups:
```
Comment (SylwiaBrant) [inline config/rbac/role.yaml:8]: From down there this file can be skipped as it was also autogenerated - reason for changes.

**Review by antoniaklja — APPROVED** [review]

File: chart/jenkins-operator/templates/operator.yaml [hunk 4]
```diff
@@ -30,11 +30,13 @@ spec:
               containerPort: 80
               protocol: TCP
           command:
-            - jenkins-operator
+            - /manager
           args: []
           env:
             - name: WATCH_NAMESPACE
-              value: {{ .Values.jenkins.namespace }}
```
Comment (prryb) [inline chart/jenkins-operator/templates/operator.yaml:37]: The operator should watch the namespace with Jenkins - not necessarily its own

File: controllers/jenkins_controller.go [hunk 5]
```diff
@@ -107,12 +107,23 @@ func (r *JenkinsReconciler) newJenkinsReconcilier(jenkins *v1alpha2.Jenkins) con
 	return config
 }
 
-// +kubebuilder:rbac:groups=jenkins.io,resources=jenkins,verbs=get;list;watch;create;update;patch;delete
```
Comment (prryb) [inline controllers/jenkins_controller.go:110]: Shouldn't we keep this line?

**Review by prryb — APPROVED** [review]

**Review by tomaszsek — APPROVED** [review]

**Pull request merged**